http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -317,6 +326,22 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
           clusterQuery.updateClusterStatus(existingCluster.id, ClusterStatus.Updating, now).transaction.void
         } else IO.unit
 
+        //maybeMasterMachineTypeChanged will throw an error if the appropriate flag hasn't been set to allow special update transitions
+        shouldFollowup = masterMachineTypeChanged.map(result => result.shouldFollowup).getOrElse(false)
+        _ <- if (shouldFollowup) {
+          //we do not support follow-up transitions when the cluster is set to an updating status
+          if (!shouldUpdate.combineAll) {
```

Comment:
this is nasty, but necessary afaik. It is needed because, if the cluster is resized (aka `shouldUpdate.combineAll` is true), then this function sets the cluster status to updating while it processes.

Instead of further complicating this complicated transition, we choose to only respect the resize, and ignore the request to update the machineType. This may not be too common a use case in the wild when you slice into what users are doing. A metric is included in the else clause to track if it is more common than we think.